src/update_engine/kernel_verifier_action.cc

```diff
@@ -24,6 +24,15 @@ void KernelVerifierAction::PerformAction() {
   }
   install_plan_ = GetInputObject();
 
+  // If unset then a new kernel was never installed.
+  if (install_plan_.new_kernel_size == 0 &&
+      install_plan_.new_kernel_hash.empty()) {
```

Comment:
Can these ever not both be either true or false? And if so, is it an error? Just spitballing some conditionals.